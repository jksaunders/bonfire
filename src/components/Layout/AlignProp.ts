import { LayoutProps } from './Layout';

export type AlignCombinations =
  | '_'
  | 'center'
  | 'start'
  | 'flex-end'
  | 'flex-start'
  | 'end'
  | 'content-center'
  | 'content-start'
  | 'content-flex-end'
  | 'content-flex-start'
  | 'content-end'
  | 'content-space-between'
  | 'content-space-around'
  | 'content-space-evenly'
  | 'content-stretch'
  | 'center _'
  | 'center center'
  | 'center start'
  | 'center flex-end'
  | 'center flex-start'
  | 'center end'
  | 'center content-center'
  | 'center content-start'
  | 'center content-flex-end'
  | 'center content-flex-start'
  | 'center content-end'
  | 'center content-space-between'
  | 'center content-space-around'
  | 'center content-space-evenly'
  | 'center content-stretch'
  | 'start _'
  | 'start center'
  | 'start start'
  | 'start flex-end'
  | 'start flex-start'
  | 'start end'
  | 'start content-center'
  | 'start content-start'
  | 'start content-flex-end'
  | 'start content-flex-start'
  | 'start content-end'
  | 'start content-space-between'
  | 'start content-space-around'
  | 'start content-space-evenly'
  | 'start content-stretch'
  | 'flex-end _'
  | 'flex-end center'
  | 'flex-end start'
  | 'flex-end flex-end'
  | 'flex-end flex-start'
  | 'flex-end end'
  | 'flex-end content-center'
  | 'flex-end content-start'
  | 'flex-end content-flex-end'
  | 'flex-end content-flex-start'
  | 'flex-end content-end'
  | 'flex-end content-space-between'
  | 'flex-end content-space-around'
  | 'flex-end content-space-evenly'
  | 'flex-end content-stretch'
  | 'flex-start _'
  | 'flex-start center'
  | 'flex-start start'
  | 'flex-start flex-end'
  | 'flex-start flex-start'
  | 'flex-start end'
  | 'flex-start content-center'
  | 'flex-start content-start'
  | 'flex-start content-flex-end'
  | 'flex-start content-flex-start'
  | 'flex-start content-end'
  | 'flex-start content-space-between'
  | 'flex-start content-space-around'
  | 'flex-start content-space-evenly'
  | 'flex-start content-stretch'
  | 'end _'
  | 'end center'
  | 'end start'
  | 'end flex-end'
  | 'end flex-start'
  | 'end end'
  | 'end content-center'
  | 'end content-start'
  | 'end content-flex-end'
  | 'end content-flex-start'
  | 'end content-end'
  | 'end content-space-between'
  | 'end content-space-around'
  | 'end content-space-evenly'
  | 'end content-stretch'
  | 'content-center _'
  | 'content-center center'
  | 'content-center start'
  | 'content-center flex-end'
  | 'content-center flex-start'
  | 'content-center end'
  | 'content-center content-center'
  | 'content-center content-start'
  | 'content-center content-flex-end'
  | 'content-center content-flex-start'
  | 'content-center content-end'
  | 'content-center content-space-between'
  | 'content-center content-space-around'
  | 'content-center content-space-evenly'
  | 'content-center content-stretch'
  | 'content-start _'
  | 'content-start center'
  | 'content-start start'
  | 'content-start flex-end'
  | 'content-start flex-start'
  | 'content-start end'
  | 'content-start content-center'
  | 'content-start content-start'
  | 'content-start content-flex-end'
  | 'content-start content-flex-start'
  | 'content-start content-end'
  | 'content-start content-space-between'
  | 'content-start content-space-around'
  | 'content-start content-space-evenly'
  | 'content-start content-stretch'
  | 'content-flex-end _'
  | 'content-flex-end center'
  | 'content-flex-end start'
  | 'content-flex-end flex-end'
  | 'content-flex-end flex-start'
  | 'content-flex-end end'
  | 'content-flex-end content-center'
  | 'content-flex-end content-start'
  | 'content-flex-end content-flex-end'
  | 'content-flex-end content-flex-start'
  | 'content-flex-end content-end'
  | 'content-flex-end content-space-between'
  | 'content-flex-end content-space-around'
  | 'content-flex-end content-space-evenly'
  | 'content-flex-end content-stretch'
  | 'content-flex-start _'
  | 'content-flex-start center'
  | 'content-flex-start start'
  | 'content-flex-start flex-end'
  | 'content-flex-start flex-start'
  | 'content-flex-start end'
  | 'content-flex-start content-center'
  | 'content-flex-start content-start'
  | 'content-flex-start content-flex-end'
  | 'content-flex-start content-flex-start'
  | 'content-flex-start content-end'
  | 'content-flex-start content-space-between'
  | 'content-flex-start content-space-around'
  | 'content-flex-start content-space-evenly'
  | 'content-flex-start content-stretch'
  | 'content-end _'
  | 'content-end center'
  | 'content-end start'
  | 'content-end flex-end'
  | 'content-end flex-start'
  | 'content-end end'
  | 'content-end content-center'
  | 'content-end content-start'
  | 'content-end content-flex-end'
  | 'content-end content-flex-start'
  | 'content-end content-end'
  | 'content-end content-space-between'
  | 'content-end content-space-around'
  | 'content-end content-space-evenly'
  | 'content-end content-stretch'
  | 'content-space-between _'
  | 'content-space-between center'
  | 'content-space-between start'
  | 'content-space-between flex-end'
  | 'content-space-between flex-start'
  | 'content-space-between end'
  | 'content-space-between content-center'
  | 'content-space-between content-start'
  | 'content-space-between content-flex-end'
  | 'content-space-between content-flex-start'
  | 'content-space-between content-end'
  | 'content-space-between content-space-between'
  | 'content-space-between content-space-around'
  | 'content-space-between content-space-evenly'
  | 'content-space-between content-stretch'
  | 'content-space-around _'
  | 'content-space-around center'
  | 'content-space-around start'
  | 'content-space-around flex-end'
  | 'content-space-around flex-start'
  | 'content-space-around end'
  | 'content-space-around content-center'
  | 'content-space-around content-start'
  | 'content-space-around content-flex-end'
  | 'content-space-around content-flex-start'
  | 'content-space-around content-end'
  | 'content-space-around content-space-between'
  | 'content-space-around content-space-around'
  | 'content-space-around content-space-evenly'
  | 'content-space-around content-stretch'
  | 'content-space-evenly _'
  | 'content-space-evenly center'
  | 'content-space-evenly start'
  | 'content-space-evenly flex-end'
  | 'content-space-evenly flex-start'
  | 'content-space-evenly end'
  | 'content-space-evenly content-center'
  | 'content-space-evenly content-start'
  | 'content-space-evenly content-flex-end'
  | 'content-space-evenly content-flex-start'
  | 'content-space-evenly content-end'
  | 'content-space-evenly content-space-between'
  | 'content-space-evenly content-space-around'
  | 'content-space-evenly content-space-evenly'
  | 'content-space-evenly content-stretch'
  | 'content-stretch _'
  | 'content-stretch center'
  | 'content-stretch start'
  | 'content-stretch flex-end'
  | 'content-stretch flex-start'
  | 'content-stretch end'
  | 'content-stretch content-center'
  | 'content-stretch content-start'
  | 'content-stretch content-flex-end'
  | 'content-stretch content-flex-start'
  | 'content-stretch content-end'
  | 'content-stretch content-space-between'
  | 'content-stretch content-space-around'
  | 'content-stretch content-space-evenly'
  | 'content-stretch content-stretch';

const parseAlignTargetValue = (
  alignProp: AlignCombinations
): { contentOrItemsKey: 'content' | 'items'; value: string } => ({
  contentOrItemsKey: alignProp.includes('content-') ? 'content' : 'items',
  value: `${alignProp}`.replace('items-', '').replace('content-', ''),
});

export const getAlignmentCss = (alignmentProp: AlignCombinations): string => {
  const [first, second] = alignmentProp.split(' ') as AlignCombinations[];
  let justifyResult = '';
  let alignResult = '';

  if (first !== '_') {
    const {
      contentOrItemsKey: firstContentOrItemsKey,
      value: firstValue,
    } = parseAlignTargetValue(first);
    justifyResult = `${firstContentOrItemsKey}: ${firstValue}`;
  }

  if (second == null) {
    alignResult = justifyResult;
  } else if (second !== '_') {
    const {
      contentOrItemsKey: secondContentOrItemsKey,
      value: secondValue,
    } = parseAlignTargetValue(second);
    alignResult = `${secondContentOrItemsKey}: ${secondValue}`;
  }

  return `
    ${alignResult && `align-${alignResult};`}
    ${justifyResult && `justify-${justifyResult};`}
  `;
};
