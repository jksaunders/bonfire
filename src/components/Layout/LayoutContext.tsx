import { createContext } from 'react';

export interface LayoutContextProps {
  defaultGap?: string;
}

const LayoutContext = createContext<LayoutContextProps>({});

export default LayoutContext;
