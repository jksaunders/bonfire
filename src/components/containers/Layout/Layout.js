import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { isArray } from "util";

const CONSTANTS = {
  JUSTIFY_CONTENT_ALIGN_ITEMS_KEYS: [
    "center",
    "start",
    "end",
    "flex-start",
    "flex-end",
    "left",
    "right",
    "normal",
    "space-between",
    "space-around",
    "space-evenly",
    "stretch"
  ],
  ORIENTATION: {
    HORIZONTAL: "horizontal",
    VERTICAL: "vertical"
  },
  SPACING: {
    SMALL: "small",
    MEDIUM: "medium",
    LARGE: "large"
  }
};

const SPACING_VALUES = {
  [CONSTANTS.SPACING.SMALL]: "4px",
  [CONSTANTS.SPACING.MEDIUM]: "8px",
  [CONSTANTS.SPACING.LARGE]: "16px",
};

const propTypes = {
  children: PropTypes.node,
  childrenFlex: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.arrayOf(PropTypes.number),
    PropTypes.func
  ]),
  childrenMargin: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.func
  ]),
  flexWrapReverse: PropTypes.bool,
  fullHeightWidth: PropTypes.bool,
  horizontalAlignment: PropTypes.oneOf(CONSTANTS.JUSTIFY_CONTENT_ALIGN_ITEMS_KEYS),
  innerContentWidth: PropTypes.string,
  linesAlignment: PropTypes.oneOf(CONSTANTS.JUSTIFY_CONTENT_ALIGN_ITEMS_KEYS),
  orientation: PropTypes.oneOf(Object.values(CONSTANTS.ORIENTATION)),
  padding: PropTypes.oneOf(Object.values(CONSTANTS.SPACING)),
  verticalAlignment: PropTypes.oneOf(CONSTANTS.JUSTIFY_CONTENT_ALIGN_ITEMS_KEYS)
};

const defaultProps = {
  children: null,
  childrenFlex: 1,
  childrenMargin: null,
  flexWrapReverse: false,
  fullHeightWidth: false,
  horizontalAlignment: "normal",
  innerContentWidth: null,
  linesAlignment: null,
  orientation: CONSTANTS.ORIENTATION.HORIZONTAL,
  padding: null,
  verticalAlignment: "normal"
};

const getFlexDirection = ({ orientation }) => {
  let direction = "";
  switch (orientation) {
    case CONSTANTS.ORIENTATION.HORIZONTAL:
      direction = "row";
      break;
    case CONSTANTS.ORIENTATION.VERTICAL:
      direction = "column";
      break;
    default: return null;
  }
  return `flex-direction: ${direction};`;
};

const getJustifyContentAndAlignItems = ({
  horizontalAlignment,
  orientation,
  verticalAlignment
}) => `
  align-items: ${orientation === CONSTANTS.ORIENTATION.HORIZONTAL ? verticalAlignment : horizontalAlignment};
  justify-content: ${orientation === CONSTANTS.ORIENTATION.HORIZONTAL ? horizontalAlignment : verticalAlignment};
`;

const getDescendentValues = ({ children, childrenFlex, childrenMargin }) => {
  if (children == null || children.length === 0) {
    return "";
  }

  const allChildrenValues = {};
  const childrenValues = [...children.map(() => ({}))];

  if (typeof childrenFlex === "number") {
    allChildrenValues.flex = childrenFlex;
  } else if (typeof childrenFlex === "function") {
    children.forEach((e, index) => {
      childrenValues[index].flex = childrenFlex(index);
    });
  } else if (isArray(childrenFlex)) {
    children.forEach((e, index) => {
      childrenValues[index].flex = childrenFlex[index];
    });
  }

  if (typeof childrenMargin === "string") {
    allChildrenValues.margin = childrenMargin;
  } else if (typeof childrenMargin === "function") {
    children.forEach((e, index) => {
      childrenValues[index].margin = childrenMargin(index);
    });
  } else if (isArray(childrenMargin)) {
    children.forEach((e, index) => {
      childrenValues[index].margin = childrenMargin[index];
    });
  }

  let result = "";
  const getSpacingValueForMargin = margin => (
    Object.values(CONSTANTS.SPACING).indexOf(margin) > -1 ? SPACING_VALUES[margin] : margin
  );

  if (Object.keys(allChildrenValues).length > 0) {
    result += `
      && > * {
        ${allChildrenValues.flex != null ? `flex: ${allChildrenValues.flex}` : ""}
        ${allChildrenValues.margin != null ? `margin: ${getSpacingValueForMargin(allChildrenValues.margin)}` : ""}
      }
    `;
  }

  childrenValues.forEach((values, index) => {
    if (Object.keys(values).length > 0) {
      result += `
        && > :nth-child(${index + 1}) {
          ${values.flex != null ? `flex: ${values.flex}` : ""}
          ${values.margin != null ? `margin: ${getSpacingValueForMargin(values.margin)}` : ""}
        }
      `;
    }
  });

  return result;
};

const getPadding = ({ padding }) => (padding != null ? `padding: ${SPACING_VALUES[padding]};` : "");

const getAlignContent = ({ linesAlignment }) => (linesAlignment != null ? `align-content: ${linesAlignment};` : "");

const Flex = styled.div`
    box-sizing: border-box;
    display: flex;
    flex-wrap: ${props => (props.flexWrapReverse ? "wrap-reverse" : "wrap")};
    height: ${props => (props.fullHeightWidth ? "100%" : "auto")};
    width: ${props => (props.fullHeightWidth ? "100%" : "auto")};

    ${getAlignContent}
    ${getDescendentValues}
    ${getFlexDirection}
    ${getJustifyContentAndAlignItems}
    ${getPadding}
`;

export const InnerContentWidthContext = React.createContext(null);

const Layout = ({
  children,
  childrenFlex,
  innerContentWidth,
  ...rest
}) => (
  <InnerContentWidthContext.Provider value={innerContentWidth}>
    <Flex childrenFlex={childrenFlex} {...rest}>
      {children}
    </Flex>
  </InnerContentWidthContext.Provider>
);

Layout.propTypes = propTypes;
Layout.defaultProps = defaultProps;
Layout.CONSTANTS = CONSTANTS;

export default Layout;