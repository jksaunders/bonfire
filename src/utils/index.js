export { default as Browser } from "./browser";
export * from "./styling";