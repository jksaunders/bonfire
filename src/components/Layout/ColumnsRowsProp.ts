import { sizesObjectToCss } from '../../utils/CssRule/CssRule';
import { LayoutProps } from './Layout';

export const getColumnsRowsCss = (direction: 'columns' | 'rows') => (
  props: LayoutProps
): string => {
  return sizesObjectToCss(props[direction], (value) => {
    const key =
      value.includes(' ') || value.includes('repeat')
        ? `grid-template-${direction}`
        : `grid-auto-${direction}`;
    return `${key}: ${value};`;
  });
};
