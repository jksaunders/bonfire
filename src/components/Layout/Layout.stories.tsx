import React from 'react';
import styled from 'styled-components';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { Layout, LayoutProps } from './Layout';

export default {
  title: 'Components/Layout',
  component: Layout,
  argTypes: {
    align: { control: 'text' },
    column: { control: 'boolean' },
    full: { control: 'boolean' },
    fullHeight: { control: 'boolean' },
    fullWidth: { control: 'boolean' },
    gap: { control: 'text' },
    row: { control: 'boolean' },
  },
} as Meta;

const Content = styled.div`
  background-color: lightblue;
  padding: 10px;
`;

const Template: Story<LayoutProps> = (args) => (
  <Layout {...args}>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
  </Layout>
);

export const Primary = Template.bind({});
Primary.args = {};
