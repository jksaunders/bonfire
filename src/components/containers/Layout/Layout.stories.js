import React from "react";
import styled from "styled-components";

import { storiesOf } from "@storybook/react";
import {
  boolean,
  select,
  text,
  withKnobs
} from "@storybook/addon-knobs";

import Layout from "./Layout";
import Typography from "../../Typography";

const StoryWrapper = styled.div`
  background-color: lightblue;
  height: ${props => props.size};
  width: ${props => props.size};
`;

const Content = styled.div`
  background-color: rgba(255, 255, 255, 0.7);
`;

storiesOf("Layout", module)
  .addDecorator(withKnobs)
  .add("basic", () => {
    const containerSize = text("Size of container", "500px");
    const flexWrapReverse = boolean("Reverse flex-wrap?", false);
    const orientation = select("Orientation", Object.values(Layout.CONSTANTS.ORIENTATION), Object.values(Layout.CONSTANTS.ORIENTATION)[0]);
    const horizontalAlignment = select("Horizontal Alignment", Layout.CONSTANTS.JUSTIFY_CONTENT_ALIGN_ITEMS_KEYS, "normal");
    const verticalAlignment = select("Vertical Alignment", Layout.CONSTANTS.JUSTIFY_CONTENT_ALIGN_ITEMS_KEYS, "normal");

    return (
      <StoryWrapper size={containerSize}>
        <Layout
          flexWrapReverse={flexWrapReverse}
          horizontalAlignment={horizontalAlignment}
          orientation={orientation}
          verticalAlignment={verticalAlignment}
        >
          <Content><Typography.H2>content1</Typography.H2></Content>
          <Content><Typography.H2>content2</Typography.H2></Content>
          <Content><Typography.H2>content3</Typography.H2></Content>
        </Layout>
      </StoryWrapper>
    );
  }).add("nested", () => {
    const containerSize = text("Size of container", "500px");
    const flexWrapReverse = boolean("Reverse flex-wrap?", false);
    const varyingFlexValues = boolean("Varying flex values?", false);
    const orientation = select("Orientation", Object.values(Layout.CONSTANTS.ORIENTATION), Object.values(Layout.CONSTANTS.ORIENTATION)[0]);
    const horizontalAlignment = select("Horizontal Alignment", Layout.CONSTANTS.JUSTIFY_CONTENT_ALIGN_ITEMS_KEYS, "normal");
    const verticalAlignment = select("Vertical Alignment", Layout.CONSTANTS.JUSTIFY_CONTENT_ALIGN_ITEMS_KEYS, "normal");
    const linesAlignment = select("Lines Alignment", Layout.CONSTANTS.JUSTIFY_CONTENT_ALIGN_ITEMS_KEYS, "stretch");

    return (
      <StoryWrapper size={containerSize}>
        <Layout
          childrenFlex={varyingFlexValues ? i => (i === 0 ? 0 : 1) : 1}
          childrenMargin={Layout.CONSTANTS.SPACING.LARGE}
          flexWrapReverse={flexWrapReverse}
          fullHeightWidth
          horizontalAlignment={horizontalAlignment}
          linesAlignment={linesAlignment}
          orientation={orientation}
          padding={Layout.CONSTANTS.SPACING.LARGE}
          verticalAlignment={verticalAlignment}
        >
          <Content><Typography.H2>content1</Typography.H2></Content>
          <Layout orientation={Layout.CONSTANTS.ORIENTATION.VERTICAL}>
            <Content><Typography.H2>content2</Typography.H2></Content>
            <Content><Typography.H2>content3</Typography.H2></Content>
          </Layout>
        </Layout>
      </StoryWrapper>
    );
  });