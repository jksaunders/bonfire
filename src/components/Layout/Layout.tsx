import React, { FC, useContext } from 'react';
import styled from 'styled-components';
import { cssRule } from '../../utils/CssRule/CssRule';
import { ResponsiveProps } from '../../utils/ResponsiveProps';
import { AlignCombinations, getAlignmentCss } from './AlignProp';
import LayoutContext from './LayoutContext';

export interface BaseLayoutProps {
  align?: AlignCombinations;
  column?: boolean;
  full?: boolean;
  fullHeight?: boolean;
  fullWidth?: boolean;
  gap?: boolean | string;
  row?: boolean;
}

export type LayoutProps = ResponsiveProps<BaseLayoutProps>;

const StyledLayout = styled.div<LayoutProps>`
  display: grid;

  ${/* align */ ''}
  ${cssRule<LayoutProps>('align', null, (props) =>
    typeof props.align === 'string' ? getAlignmentCss(props.align) : ''
  )}

  ${/* size */ ''}
  ${cssRule<LayoutProps>('full', 'height', '100%')}
  ${cssRule<LayoutProps>('full', 'width', '100%')}
  ${cssRule<LayoutProps>('fullHeight', 'height', '100%')}
  ${cssRule<LayoutProps>('fullWidth', 'width', '100%')}

  ${/* grid */ ''}
  ${cssRule<LayoutProps>('gap', 'grid-gap')}
  ${cssRule<LayoutProps>(
    ['column', 'grid-auto-flow', 'row'],
    ['row', 'grid-auto-flow', 'column']
  )}
`;

export const Layout: FC<LayoutProps> = ({
  children,
  gap: gapProp,
  ...rest
}) => {
  const layoutContext = useContext(LayoutContext);

  const gap = gapProp === true ? layoutContext.defaultGap : gapProp;

  return (
    <StyledLayout gap={gap} {...rest}>
      {children}
    </StyledLayout>
  );
};
