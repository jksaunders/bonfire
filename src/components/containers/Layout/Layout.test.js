import React from "react";
import { mount } from "enzyme";
import { expectSnapshot } from "../../../utils/snapshot";
import Layout from "./Layout";

const getLayout = (props = {}) => <Layout {...props} />;

test("Layout mounts", () => {
  const wrapper = mount(getLayout({}));
  expect(wrapper).toExist();
  expectSnapshot(getLayout({}));
});