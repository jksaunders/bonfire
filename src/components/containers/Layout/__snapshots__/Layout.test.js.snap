// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout mounts 1`] = `
<div
  className="sc-bdVaJa eTFvtk"
  orientation="horizontal"
/>
`;
